packages/vscode-extension/src/webview/providers/UtilsProvider.tsx

```diff
@@ -34,6 +36,36 @@ export function useUtils() {
   return context;
 }
 
+type TelemetryContextProps = { telemetryEnabled: boolean };
+
+const TelemetryContext = createContext<TelemetryContextProps>({ telemetryEnabled: false });
+
+export function TelemetryProvider({ children }: PropsWithChildren) {
+  const [telemetryEnabled, setTelemetryEnabled] = useState(false);
+
+  useEffect(() => {
+    utils.isTelemetryEnabled().then(setTelemetryEnabled);
+    utils.addListener("telemetryEnabledChanged", setTelemetryEnabled);
+
+    return () => {
+      utils.removeListener("telemetryEnabledChanged", setTelemetryEnabled);
+    };
+  }, []);
+
+  return (
+    <TelemetryContext.Provider value={{ telemetryEnabled }}>{children}</TelemetryContext.Provider>
```

Comment:
`{telemetryEnabled}` is useMemo -> if the value doesn't change, we'll rerender descend components anyway, cause the object will be recreated.